components/brave_mobile_subscription/renderer/android/subscription_render_frame_observer.cc

```diff
@@ -63,33 +85,60 @@ void VpnRenderFrameObserver::DidCreateScriptContext(
 
   auto connected = EnsureConnected();
   if (!connected) {
-    LOG(ERROR) << "Failed to establish connection to BraveVpnService";
+    LOG(ERROR) << "Failed to establish connection to a mojo channel";
     return;
   }
 
-#if BUILDFLAG(IS_ANDROID)
-  vpn_service_->GetPurchaseToken(base::BindOnce(
-      &VpnRenderFrameObserver::OnGetPurchaseToken, weak_factory_.GetWeakPtr()));
+  switch (product_) {
+    case Product::kVPN:
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+      vpn_service_->GetPurchaseToken(base::BindOnce(
+          &SubscriptionRenderFrameObserver::OnGetPurchaseToken,
+          weak_factory_.GetWeakPtr()));
+#endif
+      break;
+    case Product::kLeo:
+#if BUILDFLAG(ENABLE_AI_CHAT)
+      ai_chat_subscription_->GetPurchaseToken(base::BindOnce(
+          &SubscriptionRenderFrameObserver::OnGetPurchaseToken,
+          weak_factory_.GetWeakPtr()));
 #endif
+      break;
+    default:
+      NOTREACHED();
+  }
 }
 
-void VpnRenderFrameObserver::OnGetPurchaseToken(
+void SubscriptionRenderFrameObserver::OnGetPurchaseToken(
     const std::string& purchase_token) {
   if (!IsAllowed()) {
     return;
   }
   auto* frame = render_frame();
   if (frame) {
     if (IsValueAllowed(purchase_token)) {
+      DCHECK(product_ != Product::kNone);
+      std::string receipt_var_name;
+      switch (product_) {
+        case Product::kVPN:
+          receipt_var_name = "braveVpn.receipt";
+          break;
+        case Product::kLeo:
+          receipt_var_name = "braveLeo.receipt";
+          break;
+        default:
+          NOTREACHED();
+      }
       std::u16string set_local_storage =
-          base::StrCat({u"window.localStorage.setItem(\"braveVpn.receipt\", \"",
+          base::StrCat({u"window.localStorage.setItem(\"",
```

Comment:
do the conversion only one time:
`base::UTF8ToUTF16(base::StrCat(...))`

Reply:
fixed